utopia-remix/app/routes/projects.tsx

```diff
@@ -207,6 +207,29 @@ const ProjectsPageInner = React.memo(() => {
     return activeProjects.find((p) => p.proj_id === sharingProjectId) ?? null
   }, [activeProjects, sharingProjectId])
 
+  const containerRef = React.useRef<HTMLDivElement | null>(null)
+
+  const setSelectedProjectId = useProjectsStore((store) => store.setSelectedProjectId)
+
+  // deselect all projects when clicking outside of a project
+  useEffect(() => {
+    const containerRefCurrent = containerRef.current
```

Comment:
Do we really need the useEffect? Can't we just add the event listener to deselect on selection? And remove it on deselection?

Reply:
+1, I think it's just fine to add the click handler to the project entries with a stop propagation, and a click event for deselection on the component containing the entries

Reply:
yes, generally stopping propagation is not always a good practice since you're preventing other ancestors to respond to clicks for future purposes (if they want), but here it's actually ok since no one else cares about it :) changed it to use the stop propagation method. the mousedown handler fires also on context menu clicks, so added a check that it's indeed inside the container